pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -65,30 +73,44 @@ func NewExternalAlertmanager(cfg ExternalAlertmanagerConfig, orgID int64) (*Exte
 		return nil, err
 	}
 
-	return &ExternalAlertmanager{
-		amClient:      amclient.New(transport, nil),
+	logger := log.New("ngalert.notifier.remote-alertmanager")
+	amClient := amclient.New(transport, nil)
+
+	manager := NewSender(
+		&options{
+			queueCapacity: defaultMaxQueueCapacity,
+			registerer:    r,
+			timeout:       defaultTimeout,
+		},
+		logger,
+		amClient.Alert,
+	)
+
+	return &Alertmanager{
+		amClient:      amClient,
 		httpClient:    &client,
-		log:           log.New("ngalert.notifier.external-alertmanager"),
+		log:           logger,
 		url:           cfg.URL,
 		tenantID:      cfg.TenantID,
 		orgID:         orgID,
 		defaultConfig: cfg.DefaultConfig,
+		manager:       manager,
 	}, nil
 }
 
-func (am *ExternalAlertmanager) ApplyConfig(ctx context.Context, config *models.AlertConfiguration) error {
```

Comment:
A big part of the diff is this name change.
I changed the name of the struct, `remote.ExternalAlertmanager` is redundant. The same goes for file names.